### PR TITLE
Hubs/Scopes Merge 11 - Replace `IHub` with `IScopes` in Spring 5 / Spring Boot 2 integrations

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

Follow up for https://github.com/getsentry/sentry-java/pull/3297 changing Spring 5 / Spring Boot 2 integrations.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring/src/main/java/io/sentry/spring/webflux/SentryWebFilter.java [hunk 1]
```diff
@@ -34,22 +34,24 @@
...
+    Objects.requireNonNull(scopes, "scopes are required");
+    this.sentryRequestResolver = new SentryRequestResolver(scopes);
   }
 
   @Override
   public Mono<Void> filter(
       final @NotNull ServerWebExchange serverWebExchange,
       final @NotNull WebFilterChain webFilterChain) {
-    @NotNull IHub requestHub = Sentry.cloneMainHub();
+    @NotNull IScopes requestHub = Sentry.cloneMainHub();
```
Comment (lbloder) [inline sentry-spring/src/main/java/io/sentry/spring/webflux/SentryWebFilter.java:53]: Should we rename this to something like `requestScopes`?

**Review by lbloder — COMMENTED** [review]
Comment on class `EnableSentry` refer to `Hub` instead of `Scopes`.

Also, should we rename the `SentryHubRegistrar and its methods`?

File: sentry-spring-boot/src/main/java/io/sentry/spring/boot/SentryAutoConfiguration.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ static class HubConfiguration {
     }
 
     @Bean
-    public @NotNull IHub sentryHub(
+    public @NotNull IScopes sentryHub(
```
Comment (lbloder) [inline sentry-spring-boot/src/main/java/io/sentry/spring/boot/SentryAutoConfiguration.java:117]: Should we rename this method?

**Review by lbloder — COMMENTED** [review]
Should we rename `HubConfiguration` in `SentryAutoConfiguration`?

**Review by markushi — APPROVED** [review]

**Review by markushi — APPROVED** [review]

**Comment by adinauer** [comment]
We need to test whether renaming Spring beans and config classes breaks anything. Even then user code could break by renaming them.

**Pull request merged**